dali/python/backend_impl.cc

```diff
@@ -199,15 +195,11 @@ void FillTensorFromDlPack(py::capsule capsule, SourceDataType<SrcBackend> *batch
 template <typename TensorType>
 void FillTensorFromCudaArray(const py::object object, TensorType *batch, int device_id,
                              string layout) {
-  PyObject *p_ptr = object.ptr();
-  if (!PyObject_HasAttr(p_ptr, PyUnicode_FromString("__cuda_array_interface__"))) {
-    DALI_FAIL("Provided object doesn't support cuda array interface protocol")
-  }
-  PyObject *ptr_intf = PyObject_GetAttr(p_ptr, PyUnicode_FromString("__cuda_array_interface__"));
-  if (ptr_intf == Py_None) {
+  py::dict cu_a_interface = getattr(object, "__cuda_array_interface__",
+                                    py::cast<py::none>(Py_None));
```

Comment:
`py::none()` should be equivalent with `py::cast<py::none>(Py_None)`

Reply:
Done